sparkmagic/sparkmagic/livyclientlib/command.py

```diff
@@ -42,6 +42,19 @@ def execute(self, session):
             response = session.http_client.post_statement(session.id, data)
             statement_id = response[u'id']
             output = self._get_statement_output(session, statement_id)
+        except KeyboardInterrupt as e:
+            msg =  u'Interrupted by user'
+            if statement_id >= 0:
```

Comment:
Can you check for the unexpected case first (`statement_id < 0`) and raise a more informative error message (i.e "cell interrupted before the statement was created")?

Reply:
Sure, what exception should be raised here? I have yet to fully understand the uses of different exceptions.

Reply:
Now that I think about it, this doesn't seem necessary. When the execution reaches here, the user wants to interrupt the notebook and cancel the statement on Livy. The `statement_id` being invalid means the statement submission was unsuccessful and there is no need to cancel it -- this is what the user wants in most cases and she doesn't need to be informed.